src/scheduler/pbs_bitmap.cpp

```diff
@@ -74,9 +74,9 @@ pbs_bitmap_alloc(pbs_bitmap *pbm, long num_bits)
 
 	/* shrinking bitmap, clear previously used bits */
 	if (num_bits < bm->num_bits) {
-		int i;
+		long i;
 		i = num_bits / BYTES_TO_BITS(sizeof(unsigned long)) + 1;
-		for ( ; i < bm->num_longs; i++)
+		for ( ; static_cast<unsigned long>(i) < bm->num_longs; i++)
```

Comment:
i don't understand
i has to be a signed long, since we are comparing it to -1 and it's being set to a signed long. i could cast `bm->num_longs` to signed instead?

Reply:
So if num_longs is signed, why are you casting i to an unsigned long?

Reply:
var | type
-|-
i|signed
num_longs|unsigned

They have to be the same, i can either `static_cast<unsigned long>(i)` or `static_cast<long>(bm->num_longs)`

i cannot declare `i` as `unsigned long`, for reasons above

Reply:
Then make num_longs signed.  It doesn't gain much by being unsigned.